### ISSUE TITLE
[BUG] opensearch.build plugin cannot be applied after upgrading to 2.0.0

### ISSUE THREAD

**Issue by cliu123** [opensearch-project/OpenSearch#2987]
**Describe the bug**

**To Reproduce**
Upgrading BWC test to 2.0.0([Changes](https://github.com/opensearch-project/security/compare/main...cliu123:main))

**Expected behavior**
BWC test runs

**Actual behavior**
Fails at [applying opensearch.build plugin](https://github.com/opensearch-project/security/blob/main/bwc-test/build.gradle#L34)
Failing Github Action: https://github.com/cliu123/security/runs/6085915040?check_suite_focus=true
Error:
```
* What went wrong:
A problem occurred evaluating root project 'bwc-test'.
> Failed to apply plugin 'nebula.info-basic'.
   > Cannot obtain value from provider of system property 'user.name' at configuration time.
     Use a provider returned by 'forUseAtConfigurationTime()' instead.
```

**Plugins**
Security plugin

**Comment by dblock** [comment]
```
Cannot obtain value from provider of system property 'user.name' at configuration time.
```

@cliu123 missing a git configuration somewhere?
@saratvemulapalli have you seen this before?

**Comment by saratvemulapalli** [comment]
> @saratvemulapalli have you seen this before?
Haven't really seen this before. 
Initial looks seems like some where `user.name` System property is being accessed. 

@cliu123 where is `user.name` used ?

**Comment by cliu123** [comment]
> missing a git configuration somewhere?

That was my initial suspicion. But `user.name` is set in git config. And if without the git config, DCO would fail.

**Comment by cliu123** [comment]
> where is user.name used ?

Based on the error, it looks like being used in nebula.info-basic plugin that is a part/dependency of opensearch.build plugin?

**Comment by cliu123** [comment]
[This](https://github.com/opensearch-project/OpenSearch/search?q=nebula.info-basic) is the only place where `nebula.info-basic` is used.

**Comment by VachaShah** [comment]
There was a version upgrade for the nebula gradle info plugin https://github.com/opensearch-project/OpenSearch/commit/f68c8f452704f46ece805958f8d3e67d077684d8. Checking if this is related

**Comment by cliu123** [comment]
@VachaShah That looks a suspect. But how can I fix this? The [Gradle doc](https://docs.gradle.org/current/userguide/configuration_cache.html#config_cache:requirements:reading_sys_props_and_env_vars) doesn't mention any fix that can be done on our side.

**Comment by VachaShah** [comment]
I tested without the nebula info plugin upgrade. To do so:

1. Checked out the OpenSearch 2.0 branch.
2. Reverted the concerned commit.
3. `./gradlew assemble` to generate `2.0.0-SNAPSHOT` artifact on my maven local.
4. Updated [here](https://github.com/cliu123/security/blob/main/bwc-test/build.gradle#L50) and [here](https://github.com/cliu123/security/blob/main/build.gradle#L59) to `2.0.0-SNAPSHOT`.
5. Running the bwc test for security - which does not give any errors.

**Comment by cliu123** [comment]
@VachaShah Cool!. 
`2.0.0-SNAPSHOT` is older than `2.0.0-rc1-SNAPSHOT`, isn't it? But that proves that the https://github.com/opensearch-project/OpenSearch/commit/f68c8f452704f46ece805958f8d3e67d077684d8 is the cause of this error. Should we revert the commit from `2.0.0-rc1`?

**Comment by reta** [comment]
@VachaShah I do remember that update of Nebula (when I tried that while migrating to Gradke 7.3) was problematic, it also didn't work well

**Comment by VachaShah** [comment]
> @VachaShah Cool!. `2.0.0-SNAPSHOT` is older than `2.0.0-rc1-SNAPSHOT`, isn't it? But that proves that the [f68c8f4](https://github.com/opensearch-project/OpenSearch/commit/f68c8f452704f46ece805958f8d3e67d077684d8) is the cause of this error. Should we revert the commit from `2.0.0-rc1`?

I am using the latest changes from 2.0 to generate the `2.0.0-SNAPSHOT`, so I tested against my maven local. I am going to revert this nebula info plugin upgrade on `2.0` and `2.x`. And for OpenSearch `main`, open an issue with the error coming from nebula plugiin

**Comment by cliu123** [comment]
@VachaShah Thanks!

**Comment by VachaShah** [comment]
> @VachaShah I do remember that update of Nebula (when I tried that while migrating to Gradke 7.3) was problematic, it also didn't work well

Oh, @reta do you think we should not use this upgrade at all for nebula plugin (even in `main`)?

**Comment by reta** [comment]
> Oh, @reta do you think we should not use this upgrade at all for nebula plugin (even in `main`)?

@VachaShah I think it will be safer to revert it as well, thank you

**Comment by saratvemulapalli** [comment]
Thanks @VachaShah for the fixes. 
@cliu123 feel free to reopen this if you are seeing problems. Changes have been merged into 2.0.

**Comment by dbwiddis** [comment]
See comments on #3002. The root cause was a bug in Gradle 6.x which was fixed in Gradle 7; the newest version of the plugin above removed a (now deprecated graceful fallback).   TLDR:  this plugin version can be upgraded as long as Gradle 7+ is used for the build.  Why was the GHA cited above building on 6.9?
